src/graph_notebook/notebooks/04-Machine-Learning/Neptune-ML-06-Text-Encoding-Tutorial.ipynb

```diff
@@ -0,0 +1,1281 @@
+    }
+   ],
+   "source": [
+    "%%gremlin\n",
+    "g.V('user_443931').valueMap('Major')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "6cbfeb3dcf684b0ebd47d41c01317775",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Tab(children=(Output(layout=Layout(max_height='600px', overflow='scroll', width='100%')), Output(layout=Layout…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%%gremlin\n",
+    "g.V('user_443931').out('apply').hasLabel('job').valueMap('Title')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As expected, now `user_443931` has applied three jobs. And we want to provide some job recommendations based on the history job applications. Let's modify the query to predict what jobs `user_443931` is most likely to apply for.\n",
+    "\n",
+    "First, we add the `with()` step to specify the inference endpoint we want to use with our Gremlin query like this\n",
+    "`g.with(\"Neptune#ml.endpoint\",\"<INSERT ENDPOINT NAME>\")`.  \n",
+    "\n",
+    "<div style=\"background-color:#eeeeee; padding:10px; text-align:left; border-radius:10px; margin-top:10px; margin-bottom:10px; \"><b>Note</b>: The endpoint values are automatically passed into the queries below</div>\n",
+    "\n",
+    "Second, when we ask for the link within our query we use the `out()` step to predict the target node, or alternatively the `in()` step to predict the source node.  For each of these steps we need to specify the type of model being used with a `with()` clause: `with(\"Neptune#ml.prediction\")`.\n",
+    "\n",
+    "Putting these items together we get the query below, which returns 10 job titles that` user_443931` is likely to apply for."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "c00719a7203140299b53c89e96078c79",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Tab(children=(Output(layout=Layout(max_height='600px', overflow='scroll', width='100%')), Output(layout=Layout…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%%gremlin\n",
+    "g.with(\"Neptune#ml.endpoint\",\"${endpoint}\").with(\"Neptune#ml.limit\",10).\n",
+    "    V('user_443931').out('apply').with(\"Neptune#ml.prediction\").hasLabel('job').valueMap('Title')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great! Now we can see the top 10 jobs that `user_443931` is most likely to apply for. The deleted two jobs are in the top10 predictions."
```

Comment:
Can we list the top 10 results? Do they make sense? It'd be good to comment on that.

Reply:
Since the results may be slightly different for different trained models, so the listed predictions may not be the same as customer's, which may confuse them. I will list two ground truth jobs here. Then if these two jobs appear in their top 10 predictions. It indicates a good result.